client/src/components/VideoCall.jsx

```diff
@@ -48,7 +50,61 @@ export function VideoCall({ roomUrl, record }) {
         console.debug("track-started", event);
       }
     });
-  };
+
+    callFrame.on("active-speaker-change", (event) => {
+      // console.log("active speaker change");
+      const speakerEvents = player.get("speakerEvents") || [];
+      stage.set("currentSpeaker", event.activeSpeaker.peerId);
+      const timestamp = Date.now();
+      let currentCumulative = 0;
+      if (speakerEvents.length !== 0) {
+        currentCumulative = speakerEvents[speakerEvents.length-1].cumulative;
+      }
+      if (lastSpeaker === callFrame.participants().local.session_id) {
+        const speakerEvent = {
+          "type": "stop",
+          "timestamp": timestamp,
```

Comment:
timestamp may have some synchronization issues because it takes the time from the users' local clock. Can we take the time from the stage timer, instead? There is an example of this in `/client/src/components/Layouts.jsx: ElementConditionalRender`. This will also have the advantage of being tied more closely to the timing in the video.

Reply:
I've found `useVideoTime()` [link](https://docs.daily.co/reference/vcs/custom-react-hooks/use-video-time). This is under Daily's Video Component System though. Are we able to incorporate this?